docs/using-workflows/adding-triggers.md

```diff
@@ -0,0 +1,92 @@
+
+To add a push token to your workflow, configure a `trigger` section like this:
+
+```yaml
+triggers:
+- name: my-push-trigger
+  source:
+    type: push
+```
+
+When you save this workflow through the web editor or visit its page in the web app if you've updated it via CLI, you'll be presented with the authentication token under the "Settings" sidebar.
+
+![Expand the Settings sidebar to display the JWT for copying to clipboard](../images/adding-triggers-push.gif)
+
+Use this token to authenticate your HTTP requests to `api.relay.sh/api/events` with an `Authorization: Bearer <TOKEN>` header. The payload of the request should be a JSON document with a single top-level key named `data`; its values are the fields you'll extract in the `binding` section of the trigger definition. For example, a curl command would look like:
+
+```shell
+export TOKEN=... # get this from the web app
+curl -X POST -H "Authorization: Bearer $TOKEN" \
+   -d '{"data": {"message": "This is a push event"}}' \
+   https://api.relay.sh/api/events
+```
+
+## Webhook triggers
+
+Webhooks allow an external service to generate and send messages into Relay. Configuring a webhook has three parts: setting up a `trigger` container image on Relay, incorporating that image into your workflow, and configuring the external service.
+
+Relay will route incoming webhook requests to their associated trigger container. It's the container entrypoint's job to handles the incoming HTTP request and emit an event to the Relay service API to start the workflow itself. The ["Integrating with Relay"](../integrating-with-relay.md#writing-entrypoint-code) documentation has details for writing trigger entrypoints and the [relay-github](https://github.com/relay-integrations/relay-github) integration has a great working example of a webhook trigger container.
+
+Once you've found or created a trigger container, you'll need to reference it in your workflow's `triggers` section. The `source` map should set `type: webhook` and contain an `image` field, whose value is the registry path to the container image. As with `push` triggers, the trigger definition also needs a `binding` section to map the output from entrypoint's API event onto workflow parameters.
+
+The flow of data through the system is the trickiest part of webhooks. Walking through it end-to-end:
+
+* The container entrypoint parses and extracts data from the webhook payload.
+* If the data pass validation and the workflow ought to run, the entrypoint code creates a Relay event. An event is a data structure whose field names and values Relay's service API makes available to the workflow.
+* The workflow trigger definition's `binding` section maps the content of the event to workflow parameters. The keys inside the binding are the names of the parameters and the values use `!Data <fieldname>` to extract data from the event.
```

Comment:
I think it might be worth providing `binding` examples for each of the trigger types too -- showing how the data flows instead of telling, as it were.

Reply:
hm, i have more complete examples on the reference page and left them off because it felt like this was already duplicating a lot of info, wdyt https://relay.sh/docs/reference/relay-workflows/#push

Reply:
I'd definitely include a link to that in the body here but otherwise I think that's probably detailed enough for now!